### PR TITLE
Wireframe

### PR DESCRIPTION
**Pull request by AlexHamn**
Changes in this branch:
- Created a collection that keeps a list of books.
- Created a function to add a new book to the collection, with title and author.
- Create a function to remove a book from the collection.
- Display all books saved in the collection in the top part of the page.
- When a user clicks the "Add" button a new book is added to the collection and the new book is displayed on the page.
- When a user clicks the "Remove" button the correct book is removed from the collection and the correct book disappears from the page.
- Data is preserved in the browser's memory by using localStorage.

### CODE REVIEW

**Comment by AlexHamn** [comment]
The Stylelint linter fails because there is no CSS file yet.

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @AlexHamn**

### **Project Invalid ❌❌**

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, you have an error in the **`StyleLint`** and the requirement says that you shouldn't have errors.
lll


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: script.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    li.innerHTML = `
+    <h2>${book.title}</h2>  
+    <p>${book.author}</p>  
+    <button class="remove">remove</button>
+    <hr>`;
+    booksList.append(li);
+  });
+}
+
+function addBook(e) {
```
Comment (rloterh) [inline script.js:35]: - [x] Your `addBook`  function is not working as expected. When the `add` button is clicked a new book is supposed to be added to the collection, with **title** and **author**, however when just the _book title_  or the _ book author_ alone is entered your app accepts the data. Kindly improve upon this aspect by ensuring that both _book title_ and _book author_ are entered before data is accepted.

File: script.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  li.dataset.id = book.id;
+  li.innerHTML = `
+    <h2>${book.title}</h2>  
+    <p>${book.author}</p>  
+    <button class="remove">remove</button>
+    <hr>`;
+  booksList.append(li);
+}
+
+function removeBook(elem) {
```
Comment (rloterh) [inline script.js:78]: - [x] When 2 or more books with **same** _book name_ and _book author_  are stored and then a single entry having same details is removed all other entry with the same details also gets removed. You can prevent this by ensuring that each entry has a unique `id` or check if a  duplicate exists and prevent entry

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Miguel, & Alex,

You did extremely great on this project!👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there!👍 

## Good Points 👍  
- No linter errors

## Required Changes ♻️
- [ ] _Kindly check the comments under the review._
<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: script.js (hunk 1 above)
Comment (Meltrust) [reply script.js:35]: Thanks! Added validation to prevent empty fields.

File: script.js (hunk 2 above)
Comment (Meltrust) [reply script.js:78]: Thanks! Added a function to prevent duplicate entries.

File: .github/workflows/linters.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  #     - uses: actions/checkout@v2
+  #     - uses: actions/setup-node@v1
+  #       with:
+  #         node-version: "12.x"
+  #     - name: Setup Stylelint
+  #       run: |
+  #         npm install --save-dev stylelint@13.x stylelint-scss@3.x stylelint-config-standard@21.x stylelint-csstree-validator@1.x
+  #         [ -f .stylelintrc.json ] || wget https://raw.githubusercontent.com/microverseinc/linters-config/master/html-css/.stylelintrc.json
+  #     - name: Stylelint Report
+  #       run: npx stylelint "**/*.{css,scss}"
```
Comment (carlos-ssh) [inline .github/workflows/linters.yml]: - Enable your linters. More information about this, [HERE](https://github.com/microverseinc/linters-config).
Screen Shot 2021-08-18 at 19 12 27


**You are missing a General requirement!!!!!** 
- Kindly provide information in your README file. (General information about your project, how to run it locally, how to install it, provide information about the authors, add a license).

File: script.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  books.forEach((book) => {
+    const li = document.createElement('li');
+    li.innerHTML = `
+    <h2>${book.title}</h2>  
+    <p>${book.author}</p>  
+    <button class="remove">remove</button>
+    <hr>`;
+    booksList.append(li);
+  });
+}
```
Comment (carlos-ssh) [inline script.js:24]: - Fix this error in console: 
Screen Shot 2021-08-18 at 19 09 36

**Review by carlos-ssh — CHANGES_REQUESTED** [review]
Hi @AlexHamn , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by hemantso — APPROVED** [review]
Hi @AlexHamn and @Meltrust, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

## Congratulations! 🎉 

Alt Text

Cheers and happy coding 😃

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. You can mention me (@hemant-soni-vst-au4) so I can see your comments and respond.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**